### ISSUE TITLE
Can't delete playlist

### ISSUE THREAD

**Issue by NebuPookins** [quodlibet/quodlibet#4589]
Steps to reproduce
------------------
1. In the menu, choose "Browse" -> "Playlists"
2. In the bottom left corner, click "New".
3. Give the playlist a name, e.g. "Test".
4. Right click on the playlist and select "Delete"
5. When asked "Are you sure you want to delete the playlist 'Test'?", click "Delete"

Expected Output
---------------
The playlist should be deleted.

Actual Output
-------------
Playlist remains.

Test System
-----------

### Which version of Quod Libet?
4.6.0

### Which operating system 
Archlinux

**Comment by declension** [comment]
Hi. If you do launch with `quodlibet --debug`, what output do you get when replicating this issue?

**Comment by NebuPookins** [comment]
I seem to replicate the user experience described in the original ticket fairly consistently, but the output of the logs differ for different test runs which makes me suspect there might be multiple different bugs going on that contribute to the observed behavior.

When I first attempted to replicate the issue, the logs showed...

```
D: [0063.553] XSPFBackedPlaylist._delete_file: Deleting playlist file: '/home/nebu/.config/quodlibet/playlists/Test.xspf'
W: [0063.553] XSPFBackedPlaylist._delete_file: Couldn't delete '/home/nebu/.config/quodlibet/playlists/Test.xspf' ([Errno 2] No such file or directory: '/home/nebu/.config/quodlibet/playlists/Test.xspf')
```

... and when I checked the contents of the `/home/nebu/.config/quodlibet/playlists/` directory, I could confirm that indeed this file did not exist. And yet what was strange was that I believe the directory listed two files, with names `New Playlist.xspf` and `New Playlist 1.xspf` but the UI showed two playlists, with names `New Playlist` and `Test` (`Test` being the one that I created and tried to delete).

Then, when I quit and reloaded quodlibet, the UI showed two playlists, named `New Playlist` and `New Playlist 1`, and I was able to delete both of these from the UI, and when I did, the corresponding files did indeed disappear from my `/home/nebu/.config/quodlibet/playlists/` directory.

If I start with a completely empty `/home/nebu/.config/quodlibet/playlists/` directory, the bug does not seem to occur.

But if I start with `/home/nebu/.config/quodlibet/playlists/` containing a single playlist name `New Playlist.xspf` (by which I mean, first I make sure there's a playlist named `New Playlist`, and then I quit `quodlibet`, and then I start `quodlibet` again), and then create a new playlist named `Test`, then I see in the directory:

```
> ls -a /home/nebu/.config/quodlibet/playlists/
 ./   ../  'New Playlist.xspf'   Test.xspf
```

And then when I try to delete the "Test" playlist in the UI, I visually see in the UI that the "Test" playlist did _not_ get deleted. In the logs, I see:

```
D: [0103.931] XSPFBackedPlaylist._delete_file: Deleting playlist file: '/home/nebu/.config/quodlibet/playlists/Test.xspf'
```

Note that there is no warning message about being unable to delete the file this time.

And when I check the directory:

```
> ls -a /home/nebu/.config/quodlibet/playlists/
 ./   ../  'New Playlist.xspf'
```

I do see the file is deleted.

So I think there might be at least 2 bugs going on, and I can't consistently control which one gets produced.

1. Sometimes when I create a playlist, and give it a name, the name gets updated in the UI, but the name doesn't not get updated in the file system (E.g. I try to name the playlist "Test", and it shows up as "Test" in the UI, but the file is named "New Playlist 1.xspf"). When this happens, if you try to delete the playlist, it will fail with a warning about unable to delete the file "Test.xspf". If I close and reload quodlibet, then the playlist's name will match the name from the file system, and so it will seem to the user like their playlist got magically renamed instead of deleted.
2. Other times, the new playlist file will match the name of the playlist. When I try to delete the file in the UI, it _does_ get deleted in the filesystem, but the UI shows the playlist as still present. If I quit and reload quodlibet at that point, then the deleted playlist will not appear.
